padertorch/contrib/examples/pit/model.py

```diff
@@ -0,0 +1,152 @@
+import einops
+import torch
+from torch.nn.utils.rnn import PackedSequence
+
+import padertorch as pt
+from padertorch.ops.mappings import ACTIVATION_FN_MAP
+from padertorch.summary import mask_to_image, stft_to_image
+
+
+class PermutationInvariantTrainingModel(pt.Model):
+    """
+    Implements a variant of Permutation Invariant Training [1].
+
+    An example notebook can be found here:
+    /net/vol/ldrude/share/2018-12-14_pytorch.ipynb
+
+    Check out this repository to see example code:
+    git clone git@ntgit.upb.de:scratch/ldrude/pth_bss
+
+    [1] Kolbaek 2017, https://arxiv.org/pdf/1703.06284.pdf
+
+    TODO: Input normalization
+    TODO: Batch normalization
+
+    """
+    def __init__(
+            self,
+            F=257,
+            recurrent_layers=3,
+            units=600,
+            K=2,
+            dropout_input=0.,
+            dropout_hidden=0.,
+            dropout_linear=0.,
+            output_activation='relu'
+    ):
+        """
+
+        Args:
+            F: Number of frequency bins, fft_size / 2 + 1
+            recurrent_layers:
+            units: results in `units` forward and `units` backward units
+            K: Number of output streams/ speakers
+            dropout_input: Dropout forget ratio before first recurrent layer
+            dropout_hidden: Vertical forget ratio dropout between each
+                recurrent layer
+            dropout_linear: Dropout forget ratio before first linear layer
+            output_activation: Different activations. Default is 'ReLU'.
+        """
+        super().__init__()
+
+        self.K = K
+        self.F = F
+
+        assert dropout_input <= 0.5, dropout_input
+        self.dropout_input = torch.nn.Dropout(dropout_input)
+
+        assert dropout_hidden <= 0.5, dropout_hidden
+        self.blstm = torch.nn.LSTM(
+            F,
+            units,
+            recurrent_layers,
+            bidirectional=True,
+            dropout=dropout_hidden,
+        )
+
+        assert dropout_linear <= 0.5, dropout_linear
+        self.dropout_linear = torch.nn.Dropout(dropout_linear)
```

Comment:
This could also be done by pt.module.fully_connected_stack

Reply:
As it is intended as a simple example, I think the usage of several linear layers instead of the fully connected stack should be sufficient, here